projects/storefrontlib/src/lib/cms-lib/product-carousel/product-carousel.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+
+import {
+  ProductService,
+  Product,
+  CmsProductCarouselComponent
+} from '@spartacus/core';
+
+import { fromEvent, Observable, of } from 'rxjs';
+import {
+  debounceTime,
+  map,
+  distinctUntilChanged,
+  startWith
+} from 'rxjs/operators';
+
+import { CmsComponentData } from '../../cms/components/cms-component-data';
+
+const MAX_WIDTH = 360;
+const MAX_ITEM_SIZE = 4;
+const SPEED = 250;
+@Injectable()
+export class ProductCarouselService {
+  items$: Observable<Observable<Product>[]>;
+  itemSize$: Observable<number>;
+
+  constructor(
+    protected component: CmsComponentData<CmsProductCarouselComponent>,
+    private productService: ProductService
+  ) {}
+
+  /**
+   * Maps the item codes from CMS component to an array of `Product` observables.
+   */
+  setItems(): Observable<Observable<Product>[]> {
```

Comment:
now it is a service logic so probably 'getItems' I think

Reply:
changed